tests/D2L.CodeStyle.Analyzers.Test/Specs/LoggingExecutionContextScopeBuilderAnalyzer.cs

```diff
@@ -0,0 +1,59 @@
+﻿// analyzer: D2L.CodeStyle.Analyzers.ApiUsage.Logging.LoggingExecutionContextScopeBuilderAnalyzer
+
+namespace D2L.LP.Logging.ExecutionContexts {
+	using System.Threading.Tasks;
+
+	public interface ILoggingExecutionContextScopeBuilder {
+		ILoggingExecutionContextScopeBuilder With( string key, object value );
+
+		void Run( Action action );

```

Comment:
I can't enumerate them, but I expect this pattern exists elsewhere in production code, and so this analyzer should perhaps get applied to all of them.

Reply:
I have at least 1 other place where this would be useful.  But that's better dealt with later.

Reply:
I have at least 1 other place where this would be useful.  But that's better dealt with later.